## Atividade 6 
#### Algoritmos de busca sem informação

In [28]:
# importar o arquivo busca_prof.py

import busca_largura as bl

1. Escolha duas cidades para buscar uma rota usando a implementação da busca em largura, mostre o passo a passo da busca

In [29]:
# Escolheremos as duas cidades aleatoriamente

import random as rnd

rnd.seed(1)
(estado_inicial, estado_objetivo) = rnd.sample(bl.estados_romenia, 2)

print(f"{estado_inicial['estado']} -> {estado_objetivo['estado']}")

Oradea -> Iasi


In [30]:
no_Oradea = bl.No('Oradea', 0, None, None)

problema_Oradea_Iasi = bl.Problema(bl.estados_romenia, no_Oradea, lambda no: no.estado == 'Iasi')

In [31]:
busca = bl.BuscaLargura(problema_Oradea_Iasi)
busca.fronteira

[(Oradea, 0)]

A busca começa em Oradea, com fronteira vazia
A seguir, os passos dados até chegar no objetivo

In [32]:
busca.passo_busca()
busca.fronteira

[(Zerind, 71), (Sibiu, 151)]

In [33]:
busca.passo_busca()
busca.fronteira

[(Sibiu, 151), (Arad, 146)]

In [34]:
busca.passo_busca()
busca.fronteira

[(Arad, 146), (Fagaras, 250), (Rimnicu Vilcea, 231)]

In [35]:
busca.passo_busca()
busca.fronteira

[(Fagaras, 250), (Rimnicu Vilcea, 231), (Timisoara, 264)]

In [36]:
busca.passo_busca()
busca.fronteira

[(Rimnicu Vilcea, 231), (Timisoara, 264), (Bucharest, 461)]

In [37]:
busca.passo_busca()
busca.fronteira

[(Timisoara, 264), (Bucharest, 461), (Craiova, 377), (Pitesti, 328)]

In [38]:
busca.passo_busca()
busca.fronteira

[(Bucharest, 461), (Craiova, 377), (Pitesti, 328), (Lugoj, 375)]

In [39]:
busca.passo_busca()
busca.fronteira

[(Craiova, 377), (Pitesti, 328), (Lugoj, 375), (Giurgiu, 551), (Urziceni, 546)]

In [40]:
busca.passo_busca()
busca.fronteira

[(Pitesti, 328), (Lugoj, 375), (Giurgiu, 551), (Urziceni, 546), (Drobeta, 497)]

In [41]:
busca.passo_busca()
busca.fronteira

[(Lugoj, 375), (Giurgiu, 551), (Urziceni, 546), (Drobeta, 497)]

In [42]:
busca.passo_busca()
busca.fronteira

[(Giurgiu, 551), (Urziceni, 546), (Drobeta, 497), (Mehadia, 445)]

In [43]:
busca.passo_busca()
busca.fronteira

[(Urziceni, 546), (Drobeta, 497), (Mehadia, 445)]

In [44]:
busca.passo_busca()
busca.fronteira

[(Drobeta, 497), (Mehadia, 445), (Vaslui, 688), (Hirsova, 644)]

In [45]:
busca.passo_busca()
busca.fronteira

[(Mehadia, 445), (Vaslui, 688), (Hirsova, 644)]

In [46]:
busca.passo_busca()
busca.fronteira

[(Vaslui, 688), (Hirsova, 644)]

In [47]:
busca.passo_busca()
busca.fronteira

[(Hirsova, 644), (Iasi, 780)]

In [48]:
busca.passo_busca()
busca.fronteira

[(Iasi, 780), (Eforie, 730)]

No próximo passo, ele chegará a Iasi

In [49]:
busca.passo_busca()
busca.fronteira

[(Eforie, 730)]

In [50]:
busca.passo_busca()

Busca chegou ao objetivo com sucesso


In [51]:
busca.solucao

[(Oradea, 0),
 (Sibiu, 151),
 (Fagaras, 250),
 (Bucharest, 461),
 (Urziceni, 546),
 (Vaslui, 688),
 (Iasi, 780)]

A busca encontrou um caminho entre as duas cidades com sucesso e teve um custo final de 780